.github/workflows/ci.yml

```diff
@@ -52,13 +52,25 @@ jobs:
       run: ./scripts/init.sh
 
     - name: Install tarpaulin
-      run: cargo install cargo-tarpaulin
+      run: |
+        cargo install cargo-tarpaulin --version 0.18.4 && \
+        rustup install nightly-2021-11-01 && \
+        rustup target add wasm32-unknown-unknown --toolchain nightly-2021-11-01
 
     - name: Cargo check
       run: SKIP_WASM_BUILD= cargo check --all-targets --all-features --workspace
 
     - name: Generate coverage report
-      run: cargo +nightly tarpaulin --verbose --all-features --workspace --timeout 120 --out Xml
+      run: |
+        cargo \
+          +nightly-2021-11-01 \
```

Comment:
let's replace this with a `rust-toolchain` file instead?

Reply:
If this is the same toolchain as the larger chain workspace, I agree with the comment above. [Adding `rust-toolchain.toml`](https://rust-lang.github.io/rustup/overrides.html) would make this override clearer.

Reply:
@ETeissonniere & @noisefloors, introduced `rust-toolchain.toml` to override toolchain,

Is it the expected right approach ?

Reply:
@rajesh-nodle Your approach in `rust-toolchain.xxx` is correct. Rename the file to `rust-toolchain.toml` and you can safely remove the other, plain-text toolchain file.

Good call on including the target profiles and other requirements. 👍🏻

Reply:
Thanks @Nicholas,

Sorry for confusion with two files.

Was playing with `override` flag of `actions-rs/toolchain@v1` extensions.
So ended up with some messy stuff.

@ETeissonniere  & @noisefloors  Plz share your ideas ...

Here is the proposal, just let me know does it make sense.

* Global workspace `rust-toolchain.toml` is defined as below

![](https://i.imgur.com/h6a0G3P.png)

so tha normal development process & other workflow github actions are not affected.

* In `ci.yml`, toolchain is overrided with `nightly-2021-11-01` to satisfy `tarpaulin` requirements.

![](https://i.imgur.com/2fb6H0A.png)